catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableConfig.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.migratable.impl;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CatalogMigratableConfig {
+
+    //  Pkg pvt bounds on the property values - to ensure relationships are preserved in test
+
+    static final int MAX_CARDS_PER_FILE = 50000;
+
+    static final int MAX_QUERY_PAGE_SIZE = 100000;
+
+    static final int MIN_QUERY_PAGE_SIZE = 1000;
+
+    static final int MAX_THREADS = 20;
+
+    private static final Pattern RULE_FILE_NAME = Pattern.compile("[a-zA-Z]+");
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CatalogMigratableConfig.class);
+
+    private String exportFilePrefix;
+
+    private int exportCardsPerFile;
+
+    private Path exportPath;
+
+    private int exportQueryPageSize;
+
+    private int exportThreadCount;
+
+    // Needed to bypass setter restrictions when unit testing
+    CatalogMigratableConfig(int queryPageSize) {
```

Comment:
I know I'm the one who put this in but I don't really like this constructor. Do you think we can relax the min. query page size check so it can be as low as 1 and get rid of this ctor?